demisto_sdk/commands/common/update_id_set.py

```diff
@@ -302,6 +364,90 @@ def get_layout_data(path):
     if kind:
         data['kind'] = kind
     data['path'] = path
+    data['incident_and_indicator_types'] = list(incident_indicator_types_dependency)
+    if incident_indicator_fields_dependency['fieldId']:
+        data['incident_and_indicator_fields'] = incident_indicator_fields_dependency['fieldId']
+
+    return {id_: data}
+
+
+def get_incident_field_data(path, incidents_types_list):
+    data = OrderedDict()
+    json_data = get_json(path)
+
+    id_ = json_data.get('id')
+    name = json_data.get('name', '')
+    fromversion = json_data.get('fromVersion')
+    toversion = json_data.get('toVersion')
+    pack = get_pack_name(path)
+    all_associated_types = set()
+    all_scripts = set()
+
+    associated_types = json_data.get('associatedTypes')
+    if associated_types:
+        all_associated_types = set(associated_types)
+
+    system_associated_types = json_data.get('systemAssociatedTypes')
+    if system_associated_types:
+        all_associated_types = all_associated_types.union(set(system_associated_types))
+
+    if 'all' in all_associated_types:
+        all_associated_types = [list(incident_type.keys())[0] for incident_type in incidents_types_list]
```

Comment:
don't we just need to change that to ['all'] ?

Reply:
The idea is to create the id_set with the entire information so the dependency graph will only have to traverse over the data.